### DISCUSSION TITLE
unwanted p2 directory in rcp application after installation

### DISCUSSION THREAD

**Discussion by ravibodla08** [eclipse-tycho/tycho#4095] [Q&A]
after creating an eclipse rcp application, the folder structure contains some folders like "p2" which are not required for me. Even if that is required for updates I am not interested in that, as we release new rcp application version every quarter to our customers. Any possible way how I can stop from creating these folders.

installation folder structure:
![image](https://github.com/user-attachments/assets/f409be91-2f32-4a4a-8397-b3c1ee8244f9)

The main intention is to remove those files created in the binary which are having some vulnerability constraints. If we extract these files we get unsigned installers. Need to get rid of those files in particular.

**Comment by mickaelistria** [comment]
Have you tried to remove them manually or just tweaking your pom.xml file with some extra steps to just remove the `p2` folders ? I don't think Tycho needs to provide an option for that since this is a regular filesystem/zip operation that is usually well covered by existing Maven plugins.

**Comment by laeubi** [comment]
The P2 folder is created because you are using the `org.eclipse.equinox.simpleconfigurator` in your product, what is meant to support software installation / updates.
